bot/seasons/valentines/movie_generator.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import random
+from os import environ
+from urllib import parse
+
+import discord
+from discord.ext import commands
+
+TMDB_API_KEY = environ.get('TMDB_API_KEY')
+
+log = logging.getLogger(__name__)
+
+
+class RomanceMovieFinder:
+    """
+    A cog that returns a random romance movie suggestion to a user
+    """
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.command(name="romancemovie")
+    async def romance_movie(self, ctx):
+        """
+        Randomly selects a romance movie and displays information about it
+        """
+        # selecting a random int to parse it to the page parameter
+        random_page = random.randint(0, 20)
+        # TMDB api params
+        params = {"api_key": TMDB_API_KEY,
+                  "language": "en-US",
+                  "sort_by": "popularity.desc",
+                  "include_adult": "false",
+                  "include_video": "false",
+                  "page": random_page,
+                  "with_genres": "10749"
+                  }
+        # the api request url
+        request_url = "https://api.themoviedb.org/3/discover/movie?" + parse.urlencode(params)
+        async with self.bot.http_session as session:
+            async with session.get(request_url) as resp:
+                # trying to load the json file returned from the api
+                try:
+                    data = await resp.json()
+                    # selecting random result from results object in the json file
+                    selected_movie = random.choice(data["results"])
+
+                    embed = discord.Embed(
+                        title=f":sparkling_heart: {selected_movie['title']} :sparkling_heart:",
+                        description=selected_movie["overview"],
+                    )
+                    embed.set_image(url=f"http://image.tmdb.org/t/p/w200/{selected_movie['poster_path']}")
+                    embed.add_field(name="Release date :clock1:", value=selected_movie["release_date"])
+                    embed.add_field(name="Rating :star2:", value=selected_movie["vote_average"])
+                    await ctx.send(embed=embed)
+                except KeyError:
+                    embed = discord.Embed(title='Something went wrong')
```

Comment:
This needs to be more explicit & be logged.

For future reference, the [`aiohttp.ClientResponse`](https://docs.aiohttp.org/en/stable/client_reference.html#response-object) object has a [`response`](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.status) attribute as well as a [`raise_for_status()`](https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientResponse.raise_for_status) method. Since we don't have an error handler for `aiohttp.ClientResponseError`, this is fine as-is for now.

Reply:
Logged in the debug?

Reply:
`log.debug()` is fine, yes

Reply:
Log level should be warning or error.

Reply:
I have logged it , And i'm not sure what you mean by raise_for_status() should i look into that ?

Reply:
Just use `log.warning()`

Both messages still need to be more explicit. *What* went wrong? What does a `KeyError` mean here?

Reply:
I wrote a new error message , is it good enough ?

Reply:
Looks fine, but you only changed one of them :)

Reply:
Sorry i forgot to change the second one lol